core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -650,7 +650,8 @@ object ZLayer extends ZLayerCompanionVersionSpecific {
         def mapZIO[R1 <: R, E1 >: E, B: Tag](
           k: A => ZIO[R1, E1, B]
         )(implicit tag: Tag[A], trace: Trace): Default.WithContext[R1, E1, B] =
-          fromLayer(self.layer.flatMap(a => ZLayer(k(a.get[A]))))
```

Comment:
There is a random compile error about this line, happens quite often but not all the time. I tried adding explicit type parameters to see if it makes things better.